docs/source/algorithms/MaskNonOverlappingBins-v1.rst

```diff
@@ -0,0 +1,40 @@
+
+.. algorithm::
+
+.. summary::
+
+.. relatedalgorithms::
+
+.. properties::
+
+Description
+-----------
+
+This algorithm masks the bins in ``InputWorkspace`` which lie in :math:`X` range that is not covered by ``ComparisonWorkspace``. The ``MaskPartiallyOverlapping`` flag affect the behavior with regards to bins which are partially covered by ``ComparisonWorkspace``.
+
+The algorithm currently applies the default masking weight to the bins which does not clear the data.
+
+Usage
+-----
+
+**Example - MaskNonOverlappingBins**
+
+.. testcode:: MaskNonOverlappingBinsExample
+
+   bigWS = CreateSampleWorkspace(XMin=0, XMax=20000)
+   smallWS = CreateSampleWorkspace(XMin=9000, XMax=11000)
+   masked = MaskNonOverlappingBins(bigWS, smallWS)
+   print('It is not (yet) possible to access the bin masking information in Python.')
```

Comment:
Maybe another issue should be created for this.

Reply:
To my understanding, @DTasev has been working on this. I don't know the current status however. I've opened issue #23918 to remind us to update this example once the Python bindings are there.

Reply:
I needed to extract the masked bins indices only, the actual code is [here](https://github.com/mantidproject/mantid/compare/21593_create_matrixworkspace_viewer#diff-cb70fd075c89ea66dde4ab78abc42094R1103), it adds a `ws.maskedBinsIndices(index) -> [int]` function, which in Python returns a list with all the masked bins of the spectrum. If that is something you need, it will be part of the aforementioned PR